tests/test_defects.py

```diff
@@ -151,6 +151,40 @@ def check_maskBlocks(self, inputDefects, expectedDefects):
         for boxMeas in boxesMeasured:
             self.assertIn(boxMeas, unionInputExpectedBoxes)
 
+    def test_normalize_defects(self):
+        """A test for the lsst.meas.algorithms.Defect.normalize() method.
+        """
+        defects = self.allDefectsList
+        #  First series of 1-pixel contiguous defects
+        defects.append(Box2I(corner=Point2I(15, 1), dimensions=Extent2I(1, 1)))
+        defects.append(Box2I(corner=Point2I(15, 2), dimensions=Extent2I(1, 1)))
+        defects.append(Box2I(corner=Point2I(15, 3), dimensions=Extent2I(1, 1)))
+        defects.append(Box2I(corner=Point2I(15, 4), dimensions=Extent2I(1, 1)))
+        defects.append(Box2I(corner=Point2I(15, 5), dimensions=Extent2I(1, 1)))
+
+        # Second series of 1-pixel contiguos defects
+        defects.append(Box2I(corner=Point2I(20, 11), dimensions=Extent2I(1, 1)))
+        defects.append(Box2I(corner=Point2I(20, 12), dimensions=Extent2I(1, 1)))
+        defects.append(Box2I(corner=Point2I(20, 13), dimensions=Extent2I(1, 1)))
+        defects.append(Box2I(corner=Point2I(20, 14), dimensions=Extent2I(1, 1)))
+        defects.append(Box2I(corner=Point2I(20, 15), dimensions=Extent2I(1, 1)))
+
+        # the normalize() method should replace the 10 defects above by just
+        # two defect boxes with adjusted dimensions
+        newDefects = defects.normalize(self.flatExp.getBBox())
```

Comment:
Yeah, this line shows that my comment on the other PR asking about if this should be a class method is definitely justified. I think, looking at this, it should actually not be a class method, but one that normalises the set of defects itself, but I'm certainly open to the alternative, but I don't think it can be like this.

Reply:
I'll make it a method that modifies the defects object in place.

Reply:
After the comment by Tim in the PR of `meas_algorithms` (that he is fine with the method returning an object), I'll leave the method to return a new defects object.